crates/world/src/types/binary.rs

```diff
@@ -0,0 +1,263 @@
+use byteorder::ByteOrder;
+use thiserror::Error;
+
+#[derive(Debug)]
+pub struct BinaryBuffer {
```

Comment:
Why have this custom implementation? Why not just use [`std::io::Cursor`](https://doc.rust-lang.org/std/io/struct.Cursor.html) together with [`byteorder`](https://crates.io/crates/byteorder) like everywhere else?